index/04-chap4.Rmd

```diff
@@ -0,0 +1,166 @@
+
+&nbsp;
+
+Performances of methods can be compared computing their rank for each hour. The Figure \@ref(fig:barchart) compares the three best methods :
+
+- the 2 variables with the best hourly linear correlation with temperature
+- longitude, latitude and elevation
+- solar irradiance and the 2 variables with the hourly best linear correlation with temperature
+
+This barchart corroborates the precedent graph. The method based on coordinates and elevation is widely better than the two others which are more similar but with a relevant difference. The ranks $1.5$ and $2.5$ correspond to cases where two methods have exactly the same error for a same task.
+
+```{r barchart, echo=FALSE, fig.cap="Comparison of methods by rank", out.width="100%", fig.align='center'}
+include_graphics(path = "figure/barchart.png")
+```
+
+For each hour, the equation of the model is computed. They can be extracted from the benchmark. The Table \@ref(tab:models) shows the structure of the equations in the case where explanatory variables can be different from one hour to another (lm.2bestsVar). The table also shows what are the best variables and display the related error.
+
+```{r models, echo=FALSE, fig.cap="Models with their equations"}
+knitr::kable(models.sample, caption = "Models with their equations") %>%
+  kable_styling(latex_options = "scale_down") %>%
+  row_spec(0, bold = TRUE)
+```
+
+### Visualization
+
+These models can be observed on maps. For that purpose, functions building maps have been made with ggplot2 library from R for static maps and leaflet library for interactive maps.
+
+Models built from physical stations data are applied to the 1 km² grid cells. Then, the temperature is mapped with a color palette similar to the one of RMI. Class breaks are based on quantiles of temperature values. Standard error is computed for each cell and it is shown on the map with a white layer which has different levels of transparency according to the error. A large standard error is related to an opacity and vice-versa.
+
+The Figure \@ref(fig:map) shows an output for one hour based on the method where explanatory variables are Solar irradiance and the 2 variables with the best linear correlation with temperature. To build this map, some objects are needed : an object containing data (temperature and standard error) for the grid and a spatial vector object containing boundaries of Wallonia, but also the name of the variable to display. Then, some conditions can be chosen, like the display of the layer containing error, the display of the legend for error, the way to build the legend and its classes. Some arguments enable to customize the map with titles and comments. The function is thus reusable for other usages. For example, the function builds maps spatializing hydric deficit in Wallonia.
+
+The choice I made was to use quantiles to make class breaks because that is more relevant than homogeneous breaks.
```

Comment:
Ok but why is it relevant ?? ==> https://gisgeography.com/choropleth-maps-data-classification/